### PR TITLE
feat: add other policies in CEL expressions - Part 4

### PR DESCRIPTION
**Pull request by Chandan-DK**
## Related Issue(s)
Partially addresses #891 

## Description
This PR includes the conversion of policies in the other folder to Kyverno CEL policies. Conversion of the policies is done in multiple PRs

**Policies converted in this PR:**

- [x] require-annotations
- [x] require-deployments-have-multiple-replicas
- [x] require-image-checksum
- [x] require-ingress-https
- [x] require-pod-priorityclassname
- [x] require-qos-burstable 
- [x] require-storageclass

## Checklist

- [] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

File: other-cel/require-annotations/.chainsaw-test/chainsaw-test.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - patch:
+        resource:
+          apiVersion: kyverno.io/v1
+          kind: ClusterPolicy
+          metadata:
+            name: require-annotations
+          spec:
+            validationFailureAction: Enforce
+    - assert:
+        file: chainsaw-step-01-assert-1.yaml
```
[the remark this answers is not in the document]
Comment (Chandan-DK) [reply other-cel/require-annotations/.chainsaw-test/chainsaw-test.yaml]: Done

**Review by MariamFahmy98 — COMMENTED** [review]
The test `require-qos-burstable` fails because of the following error:
```
l.go:53: | 18:07:17 | require-qos-burstable | step-01  | PATCH     | ERROR | kyverno.io/v1/ClusterPolicy @ require-qos-burstable
        === ERROR
        Operation cannot be fulfilled on clusterpolicies.kyverno.io "require-qos-burstable": the object has been modified; please apply your changes to the latest version and try again
    l.go:53: | 18:07:19 | require-qos-burstable | step-01  | PATCH     | ERROR | kyverno.io/v1/ClusterPolicy @ require-qos-burstable
        === ERROR
        Patch "https://127.0.0.1:33439/apis/kyverno.io/v1/clusterpolicies/require-qos-burstable": context deadline exceeded
    l.go:53: | 18:07:19 | require-qos-burstable | step-01  | PATCH     | ERROR | kyverno.io/v1/ClusterPolicy @ require-qos-burstable
        === ERROR
        Patch "https://127.0.0.1:33439/apis/kyverno.io/v1/clusterpolicies/require-qos-burstable": context deadline exceeded
```

File: other-cel/require-storageclass/require-storageclass.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    validate:
+      cel:
+        expressions:
+          - expression: >-
+              !has(object.spec.volumeClaimTemplates) || 
+              object.spec.volumeClaimTemplates.all(volumeClaimTemplate, 
+              has(volumeClaimTemplate.spec.storageClassName) && volumeClaimTemplate.spec.storageClassName  != '')
+            message: "StatefulSets must define a storageClassName."
+
+
```
Comment (MariamFahmy98) [inline other-cel/require-storageclass/require-storageclass.yaml:48]: There is an extra line that needs to be removed.

File: other-cel/require-ingress-https/require-ingress-https.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  - name: has-tls
+    match:
+      any:
+      - resources:
+          kinds:
+          - Ingress
+    validate:
+      cel:
+        expressions:
+          - expression: "'tls' in object.spec"
```
[the remark this answers is not in the document]
Comment (Chandan-DK) [reply other-cel/require-ingress-https/require-ingress-https.yaml]: Logically speaking, both `'tls' in object.spec` and `has(object.spec.tls)` check if the `tls` key exists in `object.spec`. But `has()` seems more readable. I have made the changes.

File: other-cel/require-storageclass/require-storageclass.yaml (hunk 2 above)
Comment (Chandan-DK) [reply other-cel/require-storageclass/require-storageclass.yaml]: Done.

**Comment by MariamFahmy98** [comment]
@Chandan-DK - Could you please resolve the conflicts?

**Review by MariamFahmy98 — APPROVED** [review]
LGTM.

**Pull request merged**